optuna/_experimental.py

```diff
@@ -28,6 +26,10 @@ def _validate_version(version: str) -> None:
         )
 
 
+def _get_docstring_indent(docstring: str) -> str:
+    return docstring.split("\n")[-1] if "\n" in docstring else ""
```

Comment:
[Question] This function seems to get the last line of the given docstring. Is it always corresponding to an indent?

Reply:
> Is it always corresponding to an indent?

Not always. There are cases where this function does not work.

```python
# the last line contains characters
def f():
    """summary

    detail"""
    pass

# the last line has a different indent level
def f():
    """summary

    detail
  """
    pass

def f():
    """summary

    detail
      """
    pass
```

Reply:
A not-so-strong drop by comment but I think it's okay assuming https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#docstrings. We could add warnings in case some of these assumptions above are broken but it also sounds like overengineering.

Reply:
Thank you for your explanation. Although `flake8` and `black` do not check such inconsistent indent levels, I think we can accept the current implementation as @hvy says.